.github/workflows/build.yml

```diff
@@ -0,0 +1,76 @@
+---
+name: build
+
+on: [push]
+
+env:
+  PIP_CACHE_DIR: ~/.cache/pip
+  PRE_COMMIT_CACHE_DIR: ~/.cache/pre-commit
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+
+      - name: Cache pip test requirements
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.PIP_CACHE_DIR }}
+          key: "${{ runner.os }}-pip-test-\
+            ${{ hashFiles('**/requirements-test.txt') }}"
+          restore-keys: |
+            ${{ runner.os }}-pip-test-
+            ${{ runner.os }}-pip-
+
+      - name: Cache pre-commit hooks
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.PRE_COMMIT_CACHE_DIR }}
+          key: "${{ runner.os }}-pre-commit-\
+            ${{ hashFiles('**/.pre-commit-config.yaml') }}"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade -r requirements-test.txt
+
+      - name: Run pre-commit on all files
+        run: pre-commit run --all-files
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+
+      - name: Cache pip test requirements
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.PIP_CACHE_DIR }}
+          key: "${{ runner.os }}-pip-test-\
+            ${{ hashFiles('**/requirements-test.txt') }}"
+          restore-keys: |
+            ${{ runner.os }}-pip-test-
+            ${{ runner.os }}-pip-
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade -r requirements-test.txt
+
+      - name: Run molecule tests
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
```

Comment:
Nothing wrong with doing this... why you made the default region a secret?

Reply:
@felddy No deep reason, other than keeping all those parameters together. If I did it in plain text I predicted @dav3r would suggest we were leaking  information. :wink:

I'm fine with doing it either way; I will adopt whatever policy the group decides.

Reply:
I was only asking to see if I should change the way I did it.  I can present some prior art of not putting `AWS_DEFAULT_REGION` in secrets:

https://github.com/cisagov/skeleton-packer/blob/develop/.github/workflows/release.yml
https://github.com/cisagov/skeleton-packer/blob/develop/.github/workflows/prerelease.yml

AMI creation is a more complicated situation where the regions usage is variable and knowing the value is important to end product.   Not saying change it.  Just saying.

Reply:
What do you think, @dav3r and @mcdonnnj?

Reply:
I'm fine either way as I see no issues with having it in the clear. However, I do think it's better if we pick which way and do it consistently.

Reply:
Until now it has been in the clear in the Travis configs:
https://github.com/cisagov/ansible-role-cyhy-runner/blob/dd5cecce629272c882d004c75a51b96ebb9518e1/.travis.yml#L10

A couple of things to note about how secrets work with GitHub Actions.  When you declare a secret, and access it in an action, the value of the secret is added to an output mask.  Anywhere the secret appears will be replaced with a `***`.  This is good.  

Until you use a common value as your secret.  I have a debug ssh step in the packer project that is toggled by the existence of a secret.  I used to have it check for the value `true`, but then everywhere true was output was replaced with `***`.  So I switched it to just a definition check so that I can just but gibberish in there and trip the toggle, while still seeing both boolean values clearly in the logs.  

https://github.com/cisagov/skeleton-packer/blob/d9264d11a2c13a318c37100fed65f7c5096a9ce7/.github/workflows/build.yml#L87-L89

Anyway, if the default region is defined as a secret, then it will be masked everywhere as well.  This could be problematic if we want to jump onto the AWS Console and hunt down the instance where the test is running.  Since secrets can't be recovered from the GUI, and the value is masked in all the log output you'll have to guess where it is running.

Seeing that I just typed all this, it's evident that I changed my mind from 🤷‍♂ to 🙅‍♂.